### PR TITLE
8277133: Dragboard contents retrieved all over again during a DND process on WebView

### PR DESCRIPTION
**Pull request by jperedadnr**
This PR improves drag and drop events on a WebView node, by generating the content from the dragboard only when the drag events starts, and not for every drag over event.

A manual test is included that shows how the drag and drop experience improves, especially when dragging online images, as the time between each consecutive drag over event drops from hundreds of ms to tens of ms or even lower.

This PR is somehow blocked by #668, as dragging some images fails on macOS.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277133](https://bugs.openjdk.java.net/browse/JDK-8277133): Dragboard contents retrieved all over again during a DND process on WebView

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/680/head:pull/680` \
`$ git checkout pull/680`

Update a local copy of the PR: \
`$ git checkout pull/680` \
`$ git pull https://git.openjdk.java.net/jfx pull/680/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 680`

View PR using the GUI difftool: \
`$ git pr show -t 680`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/680.diff">https://git.openjdk.java.net/jfx/pull/680.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.web/src/main/java/javafx/scene/web/WebView.java [hunk 1]
```diff
@@ -1282,6 +1287,8 @@ private void registerEventHandlers() {
                }
            });
         setOnDragDone(event -> {
+                mimes = null;
+                values = null;
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/java/javafx/scene/web/WebView.java:1291]: Is it possible for `OnDragDone` to be missed? If so, I wonder if it is a good idea to also set `mimes` and `values` to `null` in the `OnDragDetected` handler?

**Review by kevinrushforth — COMMENTED** [review]
The fix looks good with one question, although it's likely fine as is. The manual test looks good, too (with one minor formatting issue).

File: modules/javafx.web/src/main/java/javafx/scene/web/WebView.java (hunk 1 above)
Comment (jperedadnr) [reply modules/javafx.web/src/main/java/javafx/scene/web/WebView.java:1293]: It makes sense. Done.

File: tests/manual/web/dnd/drag.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <meta charset="UTF-8">
+    <title>Title</title>
+</head>
+<body>
+<img src="blue.png"/>
+<a href="#">
+    <img src="red.png"/>
+</a>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (jperedadnr) [reply tests/manual/web/dnd/drag.html]: Done

**Review by kevinrushforth — APPROVED** [review]

**Pull request closed without merging**